dungeon/src/contrib/systems/EventScheduler.java

```diff
@@ -0,0 +1,105 @@
+ * scheduled actions, and update the state of scheduled actions.
+ *
+ * @see IAction
+ */
+public class EventScheduler extends System {
+
+  private static final PriorityQueue<ScheduledAction> scheduledActions = new PriorityQueue<>();
+
+  /**
+   * Schedules a new action to be executed after a specified delay.
+   *
+   * <p>This method creates a new ScheduledAction with the provided action and execution time, and
+   * adds it to the list of scheduled actions. The execution time is calculated as the current time
+   * plus the provided delay.
+   *
+   * @param action The action to be executed.
+   * @param delayMillis The delay in milliseconds after which the action should be executed.
+   * @return The scheduled action that was created and added to the list.
+   */
+  public static ScheduledAction scheduleAction(IAction action, long delayMillis) {
+    long executeAt = TimeUtils.millis() + delayMillis;
+    ScheduledAction scheduledAction = new ScheduledAction(action, executeAt);
+    scheduledActions.add(scheduledAction);
+    return scheduledAction;
+  }
+
+  /**
+   * Clears all scheduled actions.
+   *
+   * <p>This method removes all actions from the list of scheduled actions. After this method is
+   * called, the list of scheduled actions will be empty.
+   */
+  public static void clear() {
+    scheduledActions.clear();
+  }
+
+  /**
+   * Cancels specific scheduled actions from the list.
+   *
+   * <p>This method removes the specified scheduled action from the list of scheduled actions. If
+   * the action is not found in the list, it will not be removed.
+   *
+   * @param scheduledAction The scheduled actions to be removed.
+   * @return The removed scheduled actions.
+   */
+  public static ScheduledAction[] cancelAction(ScheduledAction... scheduledAction) {
+    for (ScheduledAction action : scheduledAction) {
+      scheduledActions.remove(action);
+    }
+    return scheduledAction;
+  }
+
+  /**
+   * Executes the scheduled actions.
+   *
+   * <p>This method is called to process all the scheduled actions. It first gets the current time,
+   * then it starts iterating over the scheduled actions. For each scheduled action, it checks if
+   * the current time is greater than or equal to the execution time of the action. If it is, it
+   * applies the effect of the action and removes the action from the list of scheduled actions.
+   * After all actions have been processed, it ends the iteration over the scheduled actions.
+   */
+  @Override
+  public void execute() {
+    long currentTime = TimeUtils.millis();
+
+    while (!scheduledActions.isEmpty()) {
+      ScheduledAction scheduledAction = scheduledActions.peek();
+      if (currentTime < scheduledAction.executeAt) {
+        break; // No more actions to execute
+      }
+      scheduledAction.action.execute();
+      scheduledActions.poll(); // Remove the action from the queue
```

Comment:
müssen wir threadsafe sein? wenn nein und niemand hier unterbrechen kann, sollte das so gehen.

wenn die ausführung unterbrochen werden kann, dann wäre vielleicht ein `scheduledActions.remove(scheduledAction)` sicherer - hier gibts du das zu entfernende element selbst an. d.h. wenn während der unterbrechung jemand die queue ändert und es einen neuen head gibt, würde trotzdem das gerade betrachtete element entfernt.

Reply:
Ist es dann nicht trotzdem `log n` weil die Action am Anfang steht?

Reply:
vermutlich nicht. die müssen das element suchen - und je nachdem, wie das intern aufgebaut ist ... also könnte sein, dass sie es bei der suche sofort finden (dann ist es wie `poll` vom aufwand her); könnte aber auch sein, dass sie es nicht gleich finden, und dann hast du den aufwand für die suche plus den aufwand für das entfernen (bzw. im o-kalkül eben das, was dominiert).

Reply:
aber: müssen wir überhaupt rücksicht auf potentielle änderungen zwischendrin nehmen? kann dieser fall überhaupt auftreten bei uns (jetzt)? wenn nicht, würde ich es bei `poll` lassen und einen kommentar dranklemmen, dass das hier in einem multithreading-szenarion schmerzen bereiten könnte.

Reply:
Kann ich grad nicht einschätzen, mir fehlt grad aber auch die Zeit für eine gründliche Analyse/Diskussion.

Vorschlag: 
(1) Du änderst das erstmal auf `poll` zurück. 
(2) Packe bitte einen Kommentar dran, dass das nur funktioniert, wenn es zwischen dem `peek` und dem `poll` keine Unterbrechung bzw. Modifikation der Queue gibt. 
(3) Mach nochmal ein Ticket auf mit Verweis auf diese Diskussion und #1909 ... das muss einfach nochmal in Ruhe nachgedacht werden.

Reply:
@Flamtky Dann kann es hier weitergehen und Du kriegst Deinen anderen Kram zu. 

@fwatermann hatte ja auch schon mal einen Scheduler auf Thread-Basis vorgeschlagen in seiner `dungine`, das könnte man da dann gleich mal mitüberlegen (in dem neuen Ticket).